autoload/go/impl.vim

```diff
@@ -35,8 +35,14 @@ function! go#impl#Impl(...) abort
 
   let result = go#util#System(printf("%s '%s' '%s'", binpath, recv, iface))
   if go#util#ShellError() != 0
-    call go#util#EchoError(result)
-    return
+    call go#util#EchoWarning("impl failed.  trying with package path")
+
+    let s:dirname = fnameescape(expand('%:p:h'))
+    let result = go#util#System(printf("%s -dir '%s' '%s' '%s'", binpath, s:dirname, recv, iface))
+    if go#util#ShellError() != 0
+      call go#util#EchoError(result)
+      return
+    end
   endif
```

Comment:
I'm not sure why we run another if this fails. If you could give me an example why it fails I can see why you did this. Why if we always pass the -dir? What exactly does this fix?

Reply:
I think we can remove this workaround now that the `impl` PR is merged?

If you could rebase on master then I can test and merge it.